odoo/addons/base/models/ir_model.py

```diff
@@ -1819,44 +1810,45 @@ def unlink(self):
                 table = data.model.model.replace('.', '_')
 
             # double-check we are really going to delete all the owners of this schema element
-            self._cr.execute("""SELECT id from ir_model_constraint where name=%s""", [name])
-            external_ids = set(x[0] for x in self._cr.fetchall())
+            external_ids = {
+                id_ for [id_] in self.env.execute_query(SQL(
+                    """SELECT id from ir_model_constraint where name=%s""", name
+                ))
+            }
             if external_ids - ids_set:
                 # as installed modules have defined this element we must not delete it!
                 continue
 
             typ = data.type
-            if typ == 'f':
-                # test if FK exists on this table (it could be on a related m2m table, in which case we ignore it)
-                self._cr.execute("""SELECT 1 from pg_constraint cs JOIN pg_class cl ON (cs.conrelid = cl.oid)
-                                    WHERE cs.contype=%s and cs.conname=%s and cl.relname=%s""",
-                                 ('f', name, table))
-                if self._cr.fetchone():
-                    self._cr.execute(SQL(
-                        'ALTER TABLE %s DROP CONSTRAINT %s',
-                        SQL.identifier(table),
-                        SQL.identifier(name[:63]),
-                    ))
-                    _logger.info('Dropped FK CONSTRAINT %s@%s', name, data.model.model)
-
-            if typ == 'u':
-                hname = sql.make_identifier(name)
-                # test if constraint exists
+            if typ in ('f', 'u'):
+                # test if FK exists on this table
                 # Since type='u' means any "other" constraint, to avoid issues we limit to
                 # 'c' -> check, 'u' -> unique, 'x' -> exclude constraints, effective leaving
                 # out 'p' -> primary key and 'f' -> foreign key, constraints.
+                # For 'f', it could be on a related m2m table, in which case we ignore it.
                 # See: https://www.postgresql.org/docs/9.5/catalog-pg-constraint.html
-                self._cr.execute("""SELECT 1 from pg_constraint cs JOIN pg_class cl ON (cs.conrelid = cl.oid)
-                                    WHERE cs.contype in ('c', 'u', 'x') and cs.conname=%s and cl.relname=%s""",
-                                 (hname, table))
-                if self._cr.fetchone():
-                    self._cr.execute(SQL(
+                hname = sql.make_identifier(name)
+                if self.env.execute_query(SQL(
+                    """SELECT 1
+                    FROM pg_constraint cs
+                    JOIN pg_class cl
+                    ON (cs.conrelid = cl.oid)
+                    WHERE cs.contype IN %s AND cs.conname = %s AND cl.relname = %s
```

Comment:
This SELECT 1 is actually useless. PostgreSQL manages correctly empty tuple result
```suggestion
                    """
                    SELECT FROM pg_constraint cs
                    JOIN pg_class cl ON (cs.conrelid = cl.oid)
                    WHERE cs.contype IN %s AND cs.conname = %s AND cl.relname = %s
```

Reply:
Habit from other SQL dialects :)